js/ui/windowManager.js

```diff
@@ -135,6 +137,76 @@ const DisplayChangeDialog = new Lang.Class({
     },
 });
 
+const EOSShellWobbly = new Lang.Class({
+    Name: 'EOSShellWobbly',
+    Extends: EndlessShellFX.Wobbly,
+    Properties: {
+        'settings' : GObject.ParamSpec.object('settings',
+                                              'Effect GSettings',
+                                              'Wobbly Effect GSettings',
+                                              GObject.ParamFlags.CONSTRUCT_ONLY |
+                                              GObject.ParamFlags.WRITABLE,
+                                              Gio.Settings)
+    },
+
+    _init: function(params) {
+        this.parent(params);
+        this._settings = null;
+    },
+
+    set settings (val) {
+         if (val.schema_id !== 'org.gnome.shell')
+             throw Error('EOSShellWobbly: Injected schema must be of ' +
+                         'org.gnome.shell');
+
+         this._settings = val;
+
+         let binder = Lang.bind(this, function(key, prop) {
+             this._settings.bind(key, this, prop, Gio.SettingsBindFlags.GET);
+         });
+
+         /* Bind to effect properties */
+         binder('wobbly-spring-k', 'spring-k');
+         binder('wobbly-spring-friction', 'friction');
+         binder('wobbly-slowdown-factor', 'slowdown-factor');
+         binder('wobbly-object-movement-range', 'object-movement-range');
+    },
+
+    grabbedByMouse: function() {
+        if (!global.get_settings().get_boolean('wobbly-effect'))
```

Comment:
I'm not a huge fan of `global`, but I suppose it isn't going to go away. Changed.